compiler/passes/src/destructuring/statement.rs

```diff
@@ -40,72 +40,83 @@ impl StatementReconstructor for DestructuringVisitor<'_> {
+                    let stmt = AssignStatement {
+                        place: (*identifier).into(),
+                        value: (*rhs_identifier).into(),
+                        id: self.state.node_builder.next_id(),
+                        span: Default::default(),
                     }
+                    .into();
 
-                    (Statement::dummy(), statements)
-                } else {
-                    assign.value = value;
-                    (assign.into(), statements)
+                    statements.push(stmt);
                 }
+
+                return (Statement::dummy(), statements);
             }
+        }
 
-            Expression::TupleAccess(access) => {
-                // We're assigning to a tuple member. Again, Aleo VM doesn't know about tuples,
-                // so we'll need to handle this.
-                let Expression::Identifier(identifier) = &access.tuple else {
-                    panic!("SSA should have ensured this is an identifier.");
-                };
+        // We may be assigning to a tuple access followed by struct or array accesses, like
+        // `tupl.1[2u8].mem = 5u8`
+        // So we need to loop and see if we find a tuple.
 
-                let tuple_ids = self.tuples.get(&identifier.name).expect("Tuple should have been encountered.");
+        assign.value = value;
+        let mut place = &mut assign.place;
 
-                // This is the correspondig variable name of the member we're assigning to.
-                let identifier = tuple_ids[access.index.value()];
+        loop {
+            match place {
+                Expression::TupleAccess(access) => {
+                    // We're assigning to a tuple member. Again, Aleo VM doesn't know about tuples,
+                    // so we'll need to handle this.
+                    let Expression::Identifier(identifier) = &access.tuple else {
+                        panic!("SSA should have ensured this is an identifier.");
+                    };
 
-                // So just assign to the variable.
-                let assign = AssignStatement {
-                    place: Expression::Identifier(identifier),
-                    value,
-                    span: Default::default(),
-                    id: self.state.node_builder.next_id(),
+                    let tuple_ids = self.tuples.get(&identifier.name).expect("Tuple should have been encountered.");
+
+                    // This is the corresponding variable name of the member we're assigning to.
+                    let identifier = tuple_ids[access.index.value()];
+
+                    *place = identifier.into();
+
+                    return (assign.into(), statements);
                 }
-                .into();
 
-                (assign, statements)
-            }
+                Expression::ArrayAccess(access) => {
+                    // We need to investigate the array, as maybe it's inside a tuple access, like `tupl.0[1u8]`.
```

Comment:
Might be my lack of understanding, but should this be the other way around `tuple[1u8].0`?

Reply:
For `tuple.0[1u8]`, wouldn't the loop hit the first match arm then terminate?

Reply:
Ah I see, the array is `tuple.0`

Reply:
What if the LHS place was a concrete expression `([1u32], [2u32]).0`, does the compiler cover us?